patches/tock/05-nrf52840-mdk-usb-dongle.patch

```diff
@@ -0,0 +1,397 @@
+@@ -0,0 +1,10 @@
++MEMORY
++{
++  rom (rx)  : ORIGIN = 0x00001000, LENGTH = 128K
++  prog (rx) : ORIGIN = 0x00030000, LENGTH = 704K
++  ram (rwx) : ORIGIN = 0x20000000, LENGTH = 256K
++}
++
++MPU_MIN_ALIGN = 8K;
++
++INCLUDE ../../kernel_layout.ld
+diff --git a/boards/nordic/nrf52840_mdk_usb_dongle/src/io.rs b/boards/nordic/nrf52840_mdk_usb_dongle/src/io.rs
+new file mode 100644
+index 00000000..2c44dc24
+--- /dev/null
++++ b/boards/nordic/nrf52840_mdk_usb_dongle/src/io.rs
+@@ -0,0 +1,57 @@
++use core::fmt::Write;
++use core::panic::PanicInfo;
++use cortexm4;
++use kernel::debug;
++use kernel::debug::IoWrite;
++use kernel::hil::led;
++use kernel::hil::uart::{self, Configure};
++use nrf52840::gpio::Pin;
++
++use crate::PROCESSES;
++
++struct Writer {
++    initialized: bool,
++}
++
++static mut WRITER: Writer = Writer { initialized: false };
++
++impl Write for Writer {
++    fn write_str(&mut self, s: &str) -> ::core::fmt::Result {
++        self.write(s.as_bytes());
++        Ok(())
++    }
++}
++
++impl IoWrite for Writer {
++    fn write(&mut self, buf: &[u8]) {
++        let uart = unsafe { &mut nrf52840::uart::UARTE0 };
++        if !self.initialized {
++            self.initialized = true;
++            uart.configure(uart::Parameters {
++                baud_rate: 115200,
++                stop_bits: uart::StopBits::One,
++                parity: uart::Parity::None,
++                hw_flow_control: false,
++                width: uart::Width::Eight,
++            });
++        }
++        for &c in buf {
++            unsafe {
++                uart.send_byte(c);
++            }
++            while !uart.tx_ready() {}
++        }
++    }
++}
++
++#[cfg(not(test))]
++#[no_mangle]
++#[panic_handler]
++/// Panic handler
++pub unsafe extern "C" fn panic_fmt(pi: &PanicInfo) -> ! {
++    // The nRF52840 Dongle LEDs (see back of board)
++    const LED1_PIN: Pin = Pin::P0_06;
++    let led = &mut led::LedLow::new(&mut nrf52840::gpio::PORT[LED1_PIN]);
++    let writer = &mut WRITER;
++    debug::panic(&mut [led], writer, pi, &cortexm4::support::nop, &PROCESSES)
++}
+diff --git a/boards/nordic/nrf52840_mdk_usb_dongle/src/main.rs b/boards/nordic/nrf52840_mdk_usb_dongle/src/main.rs
+new file mode 100644
+index 00000000..0e92473c
+--- /dev/null
++++ b/boards/nordic/nrf52840_mdk_usb_dongle/src/main.rs
+@@ -0,0 +1,149 @@
++//! Tock kernel for the Nordic Semiconductor nRF52840 dongle.
++//!
++//! It is based on nRF52840 SoC (Cortex M4 core with a BLE transceiver) with
++//! many exported I/O and peripherals.
++
++#![no_std]
++#![no_main]
++#![deny(missing_docs)]
++
++use cortexm4::scb;
++use kernel::component::Component;
++#[allow(unused_imports)]
++use kernel::{debug, debug_gpio, debug_verbose, static_init};
++use nrf52840::gpio::Pin;
++use nrf52dk_base::{SpiPins, UartChannel, UartPins};
++
++// The nRF52840 MDK USB Dongle LEDs
++const LED1_PIN: Pin = Pin::P0_08;           // not connected to a LED
```

Comment:
If it's not connected to a LED, I think the best is not to mention it at all here.